File: gems/decomplex/examples/go/function-lcom.go
package main
func mixed(price int, tax int, logger Logger) Result {
  subtotal := price + tax
  total := subtotal * 2
  rounded := total.round()

  timestamp := now()
  buffer := Buffer_init()
  buffer.push(timestamp)
  logger.info(buffer)

  return Result_init(rounded, buffer)
}

File: gems/decomplex/examples/lua/function-lcom.lua
function mixed(price, tax, logger)
  local subtotal = price + tax
  local total = subtotal * 2
  local rounded = total.round()

  local timestamp = now()
  local buffer = Buffer.init()
  local stamp = timestamp
  buffer.push(stamp)
  logger.info(buffer)

  return Result.init(rounded, buffer)
end
